ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -33,9 +78,17 @@ cpdef traceback_to_frames(traceback, max_nframes):
     while tb is not None:
         if nframes < max_nframes:
             frame = tb.tb_frame
-            code = frame.f_code
-            lineno = 0 if frame.f_lineno is None else frame.f_lineno
-            frames.insert(0, (code.co_filename, lineno, code.co_name, _extract_class_name(frame)))
+            # Python 3.11 moved PyFrameObject members to internal C API and cannot be directly accessed
+            IF PY_MAJOR_VERSION > 3 or (PY_MAJOR_VERSION == 3 and PY_MINOR_VERSION >= 11):
+                code = PyFrame_GetCode(<PyFrameObject*> frame)
```

Comment:
could frame ever be `NULL` here? does `PyFrame_GetCode` care?

should you do a similar cast as in the other function to a `PyFrameObject*` ?

Reply:
From what I understand, a traceback is always associated with a frame. @P403n1x87 can you confirm? 
I think we don't necessarily need to explicitly cast to a new `PyFrameObject*` variable here like in the other functions since we don't need to iterate through the object itself.

Reply:
This looks fine to me. I also would expect `tb_frame != NULL` if `tb` is a valid traceback object.